#! /bin/sh /usr/share/dpatch/dpatch-run
## 00_0176-coverity-fixes.diff.dpatch by Dirk Mueller <mueller@kde.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix various obvious memory leaks.

@DPATCH@
diff -urNad qt4-x11-4.3.0~/src/gui/painting/qimagescale.cpp qt4-x11-4.3.0/src/gui/painting/qimagescale.cpp
--- qt4-x11-4.3.0~/src/gui/painting/qimagescale.cpp	2007-05-25 15:24:12.000000000 +0200
+++ qt4-x11-4.3.0/src/gui/painting/qimagescale.cpp	2007-06-26 19:58:23.000000000 +0200
@@ -1005,6 +1005,7 @@
     buffer = QImage(dw, dh, src.format());
     if (buffer.isNull()) {
         qWarning("QImage: out of memory, returning null");
+        qimageFreeScaleInfo(scaleinfo);
         return QImage();
     }
 
diff -urNad qt4-x11-4.3.0~/src/svg/qsvghandler.cpp qt4-x11-4.3.0/src/svg/qsvghandler.cpp
--- qt4-x11-4.3.0~/src/svg/qsvghandler.cpp	2007-05-25 15:24:21.000000000 +0200
+++ qt4-x11-4.3.0/src/svg/qsvghandler.cpp	2007-06-26 19:58:23.000000000 +0200
@@ -1116,15 +1116,9 @@
         if (!size.isEmpty()) {
             QSvgHandler::LengthType type;
             qreal len = parseLength(size, type, handler);
-            //len = convertToPixels(len, false, type);
             // ### org_module.svg shows that font size
             // seems to be always in px...
-            type  = QSvgHandler::PX;
-            if (type == QSvgHandler::PX ||
-                type == QSvgHandler::OTHER)
-                font.setPixelSize(int(len));
-            else
-                font.setPointSizeF(len);
+            font.setPixelSize(int(len));
         }
         if (!style.isEmpty()) {
             if (style == QLatin1String("normal")) {
diff -urNad qt4-x11-4.3.0~/src/tools/uic/cpp/cppwriteicondata.cpp qt4-x11-4.3.0/src/tools/uic/cpp/cppwriteicondata.cpp
--- qt4-x11-4.3.0~/src/tools/uic/cpp/cppwriteicondata.cpp	2007-05-25 15:24:38.000000000 +0200
+++ qt4-x11-4.3.0/src/tools/uic/cpp/cppwriteicondata.cpp	2007-06-26 19:58:23.000000000 +0200
@@ -50,7 +50,7 @@
         ba[i] = r;
     }
     QByteArray ret(reinterpret_cast<const char *>(ba), baSize);
-    delete ba;
+    delete [] ba;
     return ret;
 }
 
